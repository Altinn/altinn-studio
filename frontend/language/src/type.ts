import * as nb from './nb.json';

export type TranslationKey = keyof typeof nb;
