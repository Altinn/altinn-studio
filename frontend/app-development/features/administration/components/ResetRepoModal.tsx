import React, { useEffect, useState } from 'react';
import classes from './RepoModal.module.css';
import { AltinnSpinner } from 'app-shared/components';
import { TextField } from '@altinn/altinn-design-system';
import { Button, ButtonColor, ButtonVariant } from '@digdir/design-system-react';
import { Popover } from '@mui/material';
import { getLanguageFromKey, getParsedLanguageFromKey } from 'app-shared/utils/language';
import { useAppSelector } from '../../../common/hooks';

export interface IResetRepoModalProps {
  anchorRef: React.MutableRefObject<Element>;
  onClose: any;
  open: boolean;
  language: any;
  repositoryName: string;
  handleClickResetRepo: () => void;
}

export function ResetRepoModal(props: IResetRepoModalProps) {
  const [canDelete, setCanDelete] = useState<boolean>(false);
  const [deleteRepoName, setDeleteRepoName] = useState<string>('');

  const resetting: boolean = useAppSelector((state) => state.repoStatus.resettingLocalRepo);

  useEffect(() => {
    if (deleteRepoName === props.repositoryName) {
      setCanDelete(true);
    } else {
      setCanDelete(false);
    }
  }, [deleteRepoName, props.repositoryName]);

  const onDeleteRepoNameChange = (event: any) => setDeleteRepoName(event.target.value);

  const onResetWrapper = () => {
    setCanDelete(false);
    props.handleClickResetRepo();
  };

  const onCloseWrapper = () => {
    setDeleteRepoName('');
    props.onClose();
  };
  const t = (key: string) => getLanguageFromKey(key, props.language);
  return (
    <div data-testid='reset-repo-container'>
      <Popover
        open={props.open}
        anchorEl={props.anchorRef.current}
        onClose={onCloseWrapper}
        anchorOrigin={{
          vertical: 'center',
          horizontal: 'center',
        }}
        transformOrigin={{
          vertical: 'center',
          horizontal: 'center',
        }}
        data-testid='reset-repo-popover'
      >
        <div className={classes.modalContainer}>
          <h2>{t('administration.reset_repo_confirm_heading')}</h2>
          <div>
            {getParsedLanguageFromKey(
              'administration.reset_repo_confirm_info',
              props.language,
              [props.repositoryName],
              true
            )}
          </div>
          <label htmlFor='delete-repo-name'>
            <div>{t('administration.reset_repo_confirm_repo_name')}</div>
          </label>
          <TextField id='delete-repo-name' onChange={onDeleteRepoNameChange} />
          {resetting ? (
            <AltinnSpinner />
          ) : (
            <div className={classes.buttonContainer}>
              <Button
                color={ButtonColor.Danger}
                data-testid='confirm-reset-repo-button'
                disabled={!canDelete}
                id='confirm-reset-repo-button'
                onClick={onResetWrapper}
                variant={ButtonVariant.Outline}
              >
                {t('administration.reset_repo_button')}
              </Button>
              <Button
                color={ButtonColor.Secondary}
                onClick={onCloseWrapper}
                variant={ButtonVariant.Outline}
              >
                {t('general.cancel')}
              </Button>
            </div>
          )}
        </div>
      </Popover>
    </div>
  );
}
