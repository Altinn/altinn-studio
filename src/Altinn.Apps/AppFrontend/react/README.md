## Running the tests

### Lint checks

1. Navigate to the folder `src/Altinn.Apps/AppFrontend/react`.
2. Execute `npm ci`. This step is only nescessary if you have not already done it, or if you change branches.
3. Execute `npm run install-deps`. This step is only nescessary if you have not already done it, or if you change branches.
4. Execute `npm run lint`.

### Unit tests

1. Navigate to the folder `src/Altinn.Apps/AppFrontend/react`.
2. Execute `npm ci`. This step is only nescessary if you have not already done it, or if you change branches.
3. Execute `npm run install-deps`. This step is only nescessary if you have not already done it, or if you change branches.
4. Execute `npm run test`.
