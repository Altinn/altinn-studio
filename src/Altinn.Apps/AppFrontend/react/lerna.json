{
  "command": {
    "bootstrap": {
      "hoist": true
    }
  },
  "packages": [
    "./*"
  ],
  "version": "independent"
}
