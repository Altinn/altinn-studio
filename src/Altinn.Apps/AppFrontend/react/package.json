{
	"name": "app-frontend",
	"private": true,
	"scripts": {
		"install-deps": "lerna bootstrap --hoist --ci",
		"no-ci-hoist": "lerna bootstrap --hoist --no-ci",
		"install-deps-with-lock": "lerna bootstrap",
		"test": "lerna exec --concurrency 1 -- npm test",
		"install-all": "lerna exec -- npm ci",
		"clean-lock": "lerna exec -- rm ./package-lock.json",
		"clean-node-modules": "rm -rf ./node_modules && lerna exec --concurrency 1 -- rm -rf ./node_modules",
		"update-packages": "lerna exec --concurrency 1 -- npm update --dev",
		"list-outdated": "lerna run list-outdated"
	},
	"devDependencies": {
		"@typescript-eslint/eslint-plugin": "~4.8.2",
		"@typescript-eslint/parser": "~4.8.2",
		"eslint": "~7.14.0",
		"eslint-config-airbnb": "~18.2.1",
		"eslint-plugin-import": "~2.22.1",
		"eslint-plugin-jsx-a11y": "~6.4.1",
		"eslint-plugin-react": "~7.21.5",
		"eslint-plugin-react-hooks": "^4.2.0",
		"ini": "1.3.5",
		"lerna": "3.22.1"
	}
}
