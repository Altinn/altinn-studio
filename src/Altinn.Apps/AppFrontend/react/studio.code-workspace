{
	"folders": [
		{
			"path": "."
		}
  ],
  "settings": {
    "typescript.surveys.enabled": false,
    "typescript.tsdk": "node_modules/typescript/lib",
    "eslint.alwaysShowStatus": true,
    "eslint.validate": [
      "javascript",
      "javascriptreact",
      { "language": "typescript", "autoFix": true },
      { "language": "typescriptreact", "autoFix": true}
    ]
  }
}
