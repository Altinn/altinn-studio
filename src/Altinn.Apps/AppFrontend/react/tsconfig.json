{
  "compilerOptions": {
    "outDir": "compiled",
    "module": "commonjs",
    "target": "es5",
    "lib": [
      "es6",
      "dom",
      "es7",
      "esnext"
    ],
    "sourceMap": true,
    "allowJs": true,
    "jsx": "react",
    "moduleResolution": "node",
    "rootDir": "./",
    "noImplicitReturns": true,
    "noImplicitThis": true,
    "noImplicitAny": true,
    "strictNullChecks": false,
    "noUnusedLocals": true
  },
  "include": [
    "**/*.ts",
    "**/*.tsx",
  ],
  "exclude": [
    "node_modules",
    "build",
    "scripts",
    "acceptance-tests",
    "webpack",
    "jest",
    "setupTests.ts",
    "templates",
    "**/setupTests.ts"
  ],
}
