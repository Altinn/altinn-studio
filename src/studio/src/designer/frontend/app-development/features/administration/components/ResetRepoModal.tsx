import { createMuiTheme, Grid, makeStyles, Popover, Typography } from '@material-ui/core';
import * as React from 'react';
import AltinnButton from 'app-shared/components/AltinnButton';
import AltinnInputField from 'app-shared/components/AltinnInputField';
import studioTheme from 'app-shared/theme/altinnStudioTheme';
import { getLanguageFromKey, getParsedLanguageFromKey } from 'app-shared/utils/language';
import { useSelector } from 'react-redux';
import AltinnSpinner from 'app-shared/components/AltinnSpinner';

const theme = createMuiTheme(studioTheme);

const setupClasses = makeStyles({
  sidebarHeader: {
    marginBottom: 20,
    fontSize: 20,
    fontWeight: 500,
  },
  sidebarHeaderSecond: {
    marginTop: 36,
  },
  sidebarInfoText: {
    fontSize: 16,
    marginBottom: 12,
  },
  modalContainer: {
    padding: 24,
    width: '471px',
  },
  itemSeparator: {
    paddingBottom: 12,
  },
  sectionSeparator: {
    paddingBottom: 32,
  },
  blackText: {
    color: 'black',
  },
  confirmButton: {
    backgroundColor: theme.altinnPalette.primary.red,
    color: theme.altinnPalette.primary.white,
    marginTop: 30,
  },
  cancelButton: {
    marginTop: 30,
  },
});

export interface IResetRepoModalProps {
  anchorEl: any;
  onClose: any;
  open: boolean;
  language: any;
  repositoryName: string;
  handleClickResetRepo: () => void;
}

function ResetRepoModal(props: IResetRepoModalProps) {
  const classes = setupClasses();

  const [canDelete, setCanDelete] = React.useState<boolean>(false);
  const [deleteRepoName, setDeleteRepoName] = React.useState<string>('');

  const resetting: boolean = useSelector((state: IServiceDevelopmentState) => state.repoStatus.resettingLocalRepo);

  React.useEffect(() => {
    if (deleteRepoName === props.repositoryName) {
      setCanDelete(true);
    } else {
      setCanDelete(false);
    }
  }, [deleteRepoName]);

  const onDeleteRepoNameChange = (event: any) => {
    setDeleteRepoName(event.target.value);
  };

  const onResetWrapper = () => {
    setCanDelete(false);
    props.handleClickResetRepo();
  };

  const onCloseWrapper = () => {
    setDeleteRepoName('');
    props.onClose();
  };

  return (
    <div data-testid='reset-repo-container'>
      <Popover
        open={props.open}
        anchorEl={props.anchorEl}
        onClose={onCloseWrapper}
        anchorOrigin={{
          vertical: 'center',
          horizontal: 'center',
        }}
        transformOrigin={{
          vertical: 'center',
          horizontal: 'center',
        }}
        data-testid='reset-repo-popover'
      >
        <Grid
          container={true}
          direction='column'
          className={classes.modalContainer}
        >
          <Grid item={true} className={classes.itemSeparator}>
            <Typography className={classes.sidebarHeader}>
              {getLanguageFromKey('administration.reset_repo_confirm_heading', props.language)}
            </Typography>
          </Grid>
          <Grid item={true} className={classes.sectionSeparator}>
            <Typography variant='body1'>
              {getParsedLanguageFromKey('administration.reset_repo_confirm_info', props.language, [props.repositoryName], true)}
            </Typography>
          </Grid>
          <Grid item={true}>
            <label htmlFor='delete-repo-name'>
              <Typography variant='body1' className={classes.blackText}>
                {getLanguageFromKey('administration.reset_repo_confirm_repo_name', props.language)}
              </Typography>
            </label>
          </Grid>
          <Grid item={true} className={classes.itemSeparator}>
            <AltinnInputField
              id='delete-repo-name-input'
              textFieldId='delete-repo-name'
              fullWidth={true}
              onChangeFunction={onDeleteRepoNameChange}
            />
          </Grid>
          <Grid container={true}>
            {resetting ?
              <Grid item={true} xs={6}>
                <AltinnSpinner />
              </Grid>
              :
              <>
                <Grid item={true} xs={6}>
                  <AltinnButton
                    onClickFunction={onResetWrapper}
                    btnText={getLanguageFromKey('administration.reset_repo_button', props.language)}
                    id='confirm-reset-repo-button'
                    disabled={!canDelete}
                    className={classes.confirmButton}
                    data-testid='confirm-reset-repo-button'
                  />
                </Grid>
                <Grid item={true} xs={6}>
                  <AltinnButton
                    onClickFunction={onCloseWrapper}
                    btnText={getLanguageFromKey('general.cancel', props.language)}
                    secondaryButton={true}
                    className={classes.cancelButton}
                  />
                </Grid>
              </>
            }
          </Grid>
        </Grid>
      </Popover>
    </div>
  );
}

export default ResetRepoModal;
