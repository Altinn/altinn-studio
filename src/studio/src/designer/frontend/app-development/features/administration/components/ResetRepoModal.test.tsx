import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import type { IResetRepoModalProps } from './ResetRepoModal';
import ResetRepoModal from './ResetRepoModal';
import { render, screen } from '@testing-library/react';

describe('ResetRepoModal', () => {
  let mockLanguage: any;
  let mockStore: any;
  let mockAnchorEl: any;
  const mockRepoName = 'TestRepo';
  const mockFunc = jest.fn();

  beforeEach(() => {
    const createStore = configureStore();
    const initialState: any = {
      repoStatus: {
        resetting: false,
      },
    };
    mockStore = createStore(initialState);
    mockLanguage = {
      administration: {
        reset_repo_button: 'Slett mine endringer',
        reset_repo_confirm_repo_name: 'Skriv inn repo-navn',
      },
    };
    mockAnchorEl = document.querySelector('body');
  });

  const RenderResetRepoModal = (
    props: Partial<IResetRepoModalProps>,
  ): JSX.Element => {
    const defaultProps = {
      anchorEl: mockAnchorEl,
      handleClickResetRepo: mockFunc,
      language: mockLanguage,
      onClose: mockFunc,
      open: true,
      repositoryName: mockRepoName,
    };
    return (
      <Provider store={mockStore}>
        <ResetRepoModal {...defaultProps} {...props} />
      </Provider>
    );
  };

  it('renders the component', () => {
    render(<RenderResetRepoModal />);
    const resetRepoContainer = screen.getByTestId('reset-repo-container');
    expect(resetRepoContainer).toBeDefined();
  });
});
