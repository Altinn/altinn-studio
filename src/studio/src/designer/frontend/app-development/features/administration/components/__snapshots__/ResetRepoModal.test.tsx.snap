// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResetRepoModal renders the component 1`] = `
<div
  data-testid="reset-repo-container"
/>
`;
